ethereum-explorer/src/map_contract_events.rs

```diff
@@ -1,73 +1,32 @@
-use crate::pb::eth::event::v1::Events;
 use crate::pb::eth::event::v1::Event;
-use substreams_ethereum::pb::eth::v2::Block;
-use substreams::Hex;
-use substreams_ethereum::pb::eth::v2::Log;
-use substreams_ethereum::pb::eth::v2::TransactionTrace;
-use substreams::errors::Error;
+use crate::pb::eth::event::v1::Events;
 use crate::util;
+use anyhow::Ok;
+use substreams::errors::Error;
+use substreams::Hex;
+use substreams_ethereum::pb::eth::v2::Block;
+use anyhow::anyhow;
 
 #[substreams::handlers::map]
 fn map_contract_events(contract_address: String, blk: Block) -> Result<Events, Error> {
-   let error = verify_parameter(&contract_address);
-   if error.is_some() {
-    return Err(error.unwrap());
-   }
-
-    let mut events: Vec<Event> = Vec::new();
-    let contract_address_as_vec = match Hex::decode(&contract_address) {
-        Ok(address) => address,
-        Err(error) => return Err(Error::Unexpected(error.to_string())),
-    };
-
-    for transaction in &blk.transaction_traces {
-        if transaction.to == contract_address_as_vec {
-            let transaction_events = get_transaction_events(&transaction);
-            events.extend(transaction_events);
-        }
-    }
+    verify_parameter(&contract_address)?;
+    
+    let events: Vec<Event> = blk.logs()
+            .filter(| log| log.address().to_vec() == Hex::decode(&contract_address).expect("already validated"))
+            .map(|log| Event { 
+                address: Hex::encode(log.address()), 
+                topics: log.topics().into_iter().map(| topic| Hex::encode(topic)).collect(), 
```

Comment:
```suggestion
                topics: log.topics().into_iter().map(Hex::encode).collect(), 
```

Could be used also, might be a bit too much here. Also, you still seems to have issue with the formatter. I see two spaces here before `topic`.

I tested locally the default Rustfmt rendering, your example renders better with 120, let's add file `.rustfmt.toml` in the project:

```
max_width = 120
```